course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+            scope.searchResults = [{}];
+        });
+
+        afterEach(function() {
+            // no matter what, we should end the search and clear the results.
+            expect(scope.searchInProgress).toBe(false);
+            expect(scope.searchResults).toEqual([]);
+        });
+
+        it('should alert and reload the datatable on success', function() {
+            var expectedSuccess = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            expectedSuccess.searchTerm = searchTerm;
+
+            // mock out the datatable so we can verify that it gets reloaded
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            // call it
+            scope.addUserToCourse(searchTerm, user);
+
+            // trigger the ajax calls
+            $httpBackend.expectPOST(coursePeopleURL, user)
+                        .respond(201, JSON.stringify(user)); 
+            $httpBackend.expectGET(enrollmentDetailsURL)
+                        .respond(200, JSON.stringify(enrollmentDetails));
+            $httpBackend.flush(2);
+
+            // check to see if it's reacting correctly
+            expect(scope.successes).toEqual([expectedSuccess]);
+            expect(scope.dtInstance.reloadData.calls.count()).toEqual(1);
+        });
+
+        it('should alert on a partial failure', function() {
+            var partialFailureResponse = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            partialFailureResponse.detail = 'Some kind of partial failure';
+            var expectedSuccess = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            expectedSuccess.searchTerm = searchTerm;
+            var expectedPartialFailure = {
+                searchTerm: searchTerm,
+                text: partialFailureResponse.detail,
+            };
+
+            // mock out the datatable so we can verify that it gets reloaded
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            // call it
+            scope.addUserToCourse(searchTerm, user);
+
+            // trigger the ajax calls
+            $httpBackend.expectPOST(coursePeopleURL, user)
+                        .respond(201, JSON.stringify(partialFailureResponse)); 
+            $httpBackend.expectGET(enrollmentDetailsURL)
+                        .respond(200, JSON.stringify(enrollmentDetails));
+            $httpBackend.flush(2);
+
+            // check to see if it's reacting correctly
+            expect(scope.successes).toEqual([expectedSuccess]);
+            expect(scope.partialFailures).toEqual([expectedPartialFailure]);
+            expect(scope.dtInstance.reloadData.calls.count()).toEqual(1);
+        });
+
+        it('should warn on failure to add the user', function() {
+            spyOn(scope, 'handleAjaxError');
```

Comment:
ditto -- this occurs in a couple of `it()`s, can it be in the `beforeEach()` or at the `describe()` level?